examples/sensor-websocket/src/components/SensorWebSocket.jsx

```diff
@@ -0,0 +1,206 @@
+import moment from "moment";
+
+require("moment/min/locales.min");
+const { Socket } = require("phoenix");
+
+function getLocaleFormat() {
+  const locale = window.navigator.language;
+  moment.locale(locale);
+  return moment.localeData().longDateFormat("LLL");
+}
+
+const listItem = (item, index) => {
+  return (
+    <ListGroup.Item
+      key={index}
+      className="temperature-list px-3 border-0 pb-0 py-2 font-weight-normal bg-transparent"
+    >
+      <span className="font-weight-light">
+        <b className="font-weight-bold">{item.label}: </b>
+        {item.value}
+      </span>
+    </ListGroup.Item>
+  );
+};
+
+const setSensorDataCard = (index, sensor) => {
+  const sensor_item = [
+    {
+      label: "Value",
+      value: sensor.value
+    },
+    {
+      label: "Last Update",
+      value: sensor.timestamp
+    }
+  ];
+  return (
+    <Row key={index} className="m-3 bg-light px-2 pt-3 pb-2">
+      <Col className="text-dark px-3 py-2 bg-transparent border-0 font-weight-bold w-100 text-left btn text-decoration-none">
+        <span className="bg-sensor-theme font-weight-light py-2 px-3 w-100 text-white d-inline-block">
+          <strong className="font-weight-bold">Sensor Name : &nbsp;</strong>
+          {sensor.name}
+        </span>
+        <Col className="p-0">
+          <ListGroup className="py-1 my-2">
+            {sensor_item.map(listItem)}
+          </ListGroup>
+        </Col>
+      </Col>
+    </Row>
+  );
+};
+
+const ConnectionStatus = props => {
+  const connection_type = props.connection_type;
+  return (
+    <Row className="m-3">
+      <Col className="col-12 text-dark px-3 py-2 bg-transparent border-0 w-100 text-left btn text-decoration-none">
+        <h6 className="m-0 font-weight-bold position-relative">
+          Device Status : &nbsp;
+          <span
+            className={`status-tag rounded-circle d-inline-block mr-2 ${
+              connection_type ? "connected" : "disconnected"
+            }`}
+          />
+          {connection_type ? "Connected" : "Disconnected"}
+        </h6>
+      </Col>
+    </Row>
+  );
+};
+
+class SensorWebSocket extends Component {
+  state = {
+    sensors: {},
+    device: {
+      connection_type: false
+    },
+    socket: null,
+    connected: false,
+    alerts: []
+  };
+
+  setDeviceStatus = () => {
+    const { deviceID } = this.props;
+    getDeviceDataById(deviceID).then(response => {
+      this.setState({ device: { connection_type: response.connected } });
+    });
+  };
+
+  socketDataHandler = response => {
+    if (response.event === "new_event") {
+      const { sensors } = this.state;
+      const time = response.payload.timestamp;
+      const event = response.payload.event;
+      if (event.path) {
+        sensors[event.path] = {
+          value: event.value,
+          name: event.path.split("/")[1],
+          //"/${device_id}/value".split("/") => [ "", "${device_id}", "value" ]
```

Comment:
this is a copy paste from my previous comment.
Make this developer friendly.

You can find more on that topic with a simple Google search "code comments best practices", add "javascript" for more specific examples
https://blog.risingstack.com/javascript-clean-coding-best-practices-node-js-at-scale/